addons/mass_mailing_sms/static/src/js/fields_sms_widget.js

```diff
@@ -0,0 +1,93 @@
+/** @odoo-module **/
+
+import { _t } from "web.core";
+import SmsWidget from '@sms/js/fields_sms_widget';
+
+const TEXT_URL_REGEX = /https?:\/\/[\w@:%.+&~#=/-]+(?:\?\S+)?/g  // from tools.mail.TEXT_URL_REGEX
+
+/**
+ * Override to provide extra characters count information to
+ * consider links converted with link_tracker and opt-out
+ * link if the option is selected.
+ */
+SmsWidget.include({
+    events: _.extend({}, SmsWidget.prototype.events || {}, {
+        'focusin': '_onClick',
+    }),
+    /**
+     *@override
+     */
+    init: function () {
+        this._super.apply(this, arguments);
+        if (this.model === "mailing.mailing" && this.mode === 'edit') {
+            this._rpc({
+                model: 'mailing.mailing',
+                method: 'get_chars_counts',
+                args: [this.res_id]
+            }).then(({ optOutLink, link }) => {
+                this.optOutEnabled = this.recordData.sms_sms_allow_unsubscribe;
+                this.noticeLinksReplaced = false;
+                this.charsCountData = { optOutLink, link };
+            });
+        }
+    },
+    /**
+     * Add the characters for opt-out link if enabled.
+     *
+     * @override
+     */
+    _getNbrChar: function () {
+        let nbrChar = this._super.apply(this, arguments);
+        if (this.charsCountData) {
+            // this is the simplest way to read from another field in a legacy fix.
+            const sms_allow_unsubscribe_el = document.querySelector('div[name="sms_allow_unsubscribe"] input');
```

Comment:
Not that much better, but you can also access the desired up to date information like this:

```js
const controller = this.findAncestor((ancestor) => {
    return typeof ancestor.model === 'object' && ancestor.model.loadParams && ancestor.model.loadParams.modelName === this.model
});
this.optOutEnabled = controller && controller.model.get(this.record.id).data['sms_allow_unsubscribe'];
```

Reply:
Fortunately the reviewer is the one that prefers plain hackish JS compared to convoluted 2024 JS.